test/test_convergence.jl

```diff
@@ -0,0 +1,213 @@
+
+    e_int = TD.internal_energy.(Ref(param_set), T, q_pt)
+    θ_liq_ice = TD.liquid_ice_pottemp.(Ref(param_set), T, ρ, q_pt)
+    RH = TD.relative_humidity.(Ref(param_set), T, p, Ref(phase_type), q_pt)
+    e_pot = grav * z
+    Random.seed!(15)
+    u = rand(FT, size(T)) * 5
+    v = rand(FT, size(T)) * 5
+    w = rand(FT, size(T)) * 5
+    e_kin = (u .^ 2 .+ v .^ 2 .+ w .^ 2) / 2
+
+    return ProfileSet{typeof(T), typeof(q_pt), typeof(phase_type)}(
+        z,
+        T,
+        p,
+        RS,
+        e_int,
+        ρ,
+        θ_liq_ice,
+        q_tot,
+        q_liq,
+        q_ice,
+        q_pt,
+        RH,
+        e_pot,
+        u,
+        v,
+        w,
+        e_kin,
+        phase_type,
+    )
+end
+function input_config(ArrayType; n = 10, n_RS1 = 20, n_RS2 = 20, T_surface = 290, T_min = 150)
+    n_RS = n_RS1 + n_RS2
+    z_range = ArrayType(range(0, stop = 5e2, length = n))
+    relative_sat1 = ArrayType(range(0, stop = 1, length = n_RS1))
+    relative_sat2 = ArrayType(range(1, stop = 1.02, length = n_RS2))
+    relative_sat = vcat(relative_sat1, relative_sat2)
+    return z_range, relative_sat, T_surface, T_min
+end
+
+"""
+    shared_profiles(
+        param_set::APS,
+        z_range::AbstractArray,
+        relative_sat::AbstractArray,
+        T_surface,
+        T_min,
+    )
+
+Compute profiles shared across `PhaseDry`,
+`PhaseEquil` and `PhaseNonEquil` thermodynamic
+states, including:
+ - `z` altitude
+ - `T` temperature
+ - `p` pressure
+ - `RS` relative saturation
+"""
+function shared_profiles(param_set::APS, z_range::AbstractArray, relative_sat::AbstractArray, T_surface, T_min)
+    FT = eltype(z_range)
+    n_RS = length(relative_sat)
+    n = length(z_range)
+    T = similar(z_range, n * n_RS)
+    p = similar(z_range, n * n_RS)
+    RS = similar(z_range, n * n_RS)
+    z = similar(z_range, n * n_RS)
+    linear_indices = LinearIndices((1:n, 1:n_RS))
+    # We take the virtual temperature, returned here,
+    # as the temperature, and then compute a thermodynamic
+    # state consistent with that temperature. This profile
+    # will not be in hydrostatic balance, but this does not
+    # matter for the thermodynamic test profiles.
+    profile = TP.DecayingTemperatureProfile{FT}(param_set, FT(T_surface), FT(T_min))
+    for i in linear_indices.indices[1]
+        for j in linear_indices.indices[2]
+            k = linear_indices[i, j]
+            z[k] = z_range[i]
+            T[k], p[k] = profile(param_set, z[k])
+            RS[k] = relative_sat[j]
+        end
+    end
+    return z, T, p, RS
+end
+
+profiles_sfc = []
+profiles_int = []
+profiles = PhaseEquilProfiles(thermo_params, Array{FT});
+
+## Properties contained in `profiles`
+## :z, :T, :p, :RS, :e_int, :h, :ρ, :θ_liq_ice, :q_tot, :q_liq, :q_ice, :q_pt, :RH, :e_pot, :u, :v, :w, :e_kin, :phase_type
+
+for prof in profiles
+    if prof.z == FT(0)
+        push!(profiles_sfc, prof)
+    else
+        push!(profiles_int, prof)
+    end
+end
```

Comment:
I would recommend using `filter` instead of `push!`ing to a generic container

Reply:
Done in #57